### ISSUE TITLE
Use deps/build.jl

### ISSUE THREAD

**Issue by abelsiqueira** [JuliaOptimizers/CUTEst.jl#50]
Suggested by @timholy.

**Comment by dpo** [comment]
We have releases now:
* https://github.com/optimizers/archdefs-mirror
* https://github.com/optimizers/sifdecode-mirror
* https://github.com/optimizers/cutest-mirror
* https://github.com/optimizers/mastsif-mirror
* https://github.com/optimizers/maros-meszaros-mirror

See also https://github.com/optimizers/homebrew-cutest/commits/release

**Comment by abelsiqueira** [comment]
I implemented this on my branch [feat/bindeps](https://github.com/abelsiqueira/CUTEst.jl/tree/feat/bindeps) on top of #48. I'll wait until #48 is merged to prevent a huge PR. Also, it currently does not work on OSX, but I can't test easily.
Despite the branch name, it does not use BinDeps. I tried to use it, but for a non-straightforward build it isn't very helpful. I ended up using Homebrew for OSX and my new package [linux-cutest](https://github.com/abelsiqueira/linux-cutest) for Linux.

The most interesting part is the nonnecessity of setting anything on bashrc, like is usually required.
